server/src/test/java/org/elasticsearch/snapshots/RepositoryIntegrityHealthIndicatorServiceTests.java

```diff
@@ -13,48 +13,78 @@
 import org.elasticsearch.cluster.metadata.RepositoryMetadata;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.node.DiscoveryNodeUtils;
+import org.elasticsearch.cluster.node.DiscoveryNodes;
 import org.elasticsearch.cluster.service.ClusterService;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.health.Diagnosis;
 import org.elasticsearch.health.Diagnosis.Resource.Type;
 import org.elasticsearch.health.HealthIndicatorDetails;
-import org.elasticsearch.health.HealthIndicatorImpact;
 import org.elasticsearch.health.HealthIndicatorResult;
-import org.elasticsearch.health.ImpactArea;
 import org.elasticsearch.health.SimpleHealthIndicatorDetails;
 import org.elasticsearch.health.node.HealthInfo;
+import org.elasticsearch.health.node.RepositoriesHealthInfo;
 import org.elasticsearch.test.ESTestCase;
+import org.junit.Before;
 
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.stream.Stream;
 
 import static org.elasticsearch.common.util.CollectionUtils.appendToCopy;
 import static org.elasticsearch.health.HealthStatus.GREEN;
-import static org.elasticsearch.health.HealthStatus.RED;
+import static org.elasticsearch.health.HealthStatus.YELLOW;
 import static org.elasticsearch.repositories.RepositoryData.CORRUPTED_REPO_GEN;
 import static org.elasticsearch.repositories.RepositoryData.EMPTY_REPO_GEN;
-import static org.elasticsearch.snapshots.RepositoryIntegrityHealthIndicatorService.CORRUPTED_REPOSITORY;
+import static org.elasticsearch.snapshots.RepositoryIntegrityHealthIndicatorService.CORRUPTED_DEFINITION;
+import static org.elasticsearch.snapshots.RepositoryIntegrityHealthIndicatorService.INVALID_DEFINITION;
 import static org.elasticsearch.snapshots.RepositoryIntegrityHealthIndicatorService.NAME;
+import static org.elasticsearch.snapshots.RepositoryIntegrityHealthIndicatorService.UNKNOWN_DEFINITION;
 import static org.hamcrest.Matchers.equalTo;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
 public class RepositoryIntegrityHealthIndicatorServiceTests extends ESTestCase {
 
-    public void testIsGreenWhenAllRepositoriesAreNotCorrupted() {
+    private DiscoveryNode node1;
+    private DiscoveryNode node2;
+    private HealthInfo healthInfo;
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+
+        node1 = DiscoveryNodeUtils.create(randomAlphaOfLength(10), randomUUID());
+        node2 = DiscoveryNodeUtils.create(randomAlphaOfLength(10), randomUUID());
+        healthInfo = new HealthInfo(
+            Map.of(),
+            null,
+            new HashMap<>(
+                Map.of(
+                    node1.getId(),
+                    new RepositoriesHealthInfo(List.of(), List.of()),
+                    node2.getId(),
+                    new RepositoriesHealthInfo(List.of(), List.of())
+                )
+            )
+        );
```

Comment:
I think we should initialise this in each test so it's closer to the code that uses it. What do you think?

Reply:
I implemented it this way because I needed to have a non-empty map of repositories health info to avoid the repo analyzer class complaining that the status was "unknown". Therefore, I chose to create a "green" health info before each test so every test has the same "green" starting position. Do you still think it makes more sense to move the health info initialization to the tests themselves?

Reply:
Hm, so I do not think we should fake the starting point. Should we reconsider keeping the health green when there are missing health information, to avoid having the lag upon restart?

Reply:
I copied that approach from the [DiskHealthIndicatorService.java](https://github.com/nielsbauman/elasticsearch/blob/f2552321a46ea948f2146b821c65c58d84e9fabd/server/src/main/java/org/elasticsearch/health/node/DiskHealthIndicatorService.java#L87-L87). I think it makes sense to keep the status to unknown when there is missing information (either due to startup or some other cause); that feels more accurate to me. What do you think?

Reply:
We have been going back on forth on this honestly. For DSL we decided to start with `GREEN` mainly because it would take even longer to get an answer since DSL only runs every 5 minutes.

It is more accurate but it's also more noisy, that's why you see us considering. The reason is that `unknown` is "higher" than `green` during merging so it takes over the overall status.

We can keep the unknown here too, to be inline with the disk indicator and maybe bring back the topic if we see that it's too noisy. Or maybe we should re-evaluate the strategy for merging. What do you think?